public/index.pug

```diff
@@ -125,9 +125,7 @@ html
               ) #{icon.license.type}
             div(class="grid-item__footer")
               button(class="grid-item__preview copy-button" title=`${icon.title} SVG` disabled)
-                svg(id=`${icon.slug}-svg` role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg")
-                  title #{icon.title} icon
-                  path(d=icon.path)
+                img(src=`./icons/${icon.slug}.svg` loading="lazy")
```

Comment:
Great!
Another downside that I did not think before: I think the `fill` property will not work anymore...
I think, the icon wont be filled in white in dark mode. Cannot try it right now though.

Reply:
Good point, you're correct. Thanks for pointing it out :+1: 

I think this can be solved using plain HTML, let me investigate :thinking:

Reply:
The possible alternative would be to use `img` with data url (svg encoded in base64). Something like this:
```
<img src="data:image/svg+xml;base64,PHN2ZyByb2xlPSJpbWciIHZpZXdCb3g9IjAgMCAyNCAyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48dGl0bGU+U2ltcGxlIEljb25zIGljb248L3RpdGxlPjxwYXRoIGQ9Ik0xMiAwQzguNjg4IDAgNiAyLjY4OCA2IDZzMi42ODggNiA2IDZjMS4wNjYgMCAyLjExNzEuMjg2MiAzLjAzNzEuODI2MiAyLjg1OCAxLjY3OCAzLjgxNjcgNS4zNTM5IDIuMTM4NyA4LjIxMjloMS42Nzk3YTcuNDkyNSA3LjQ5MjUgMCAwIDAgLjY0MjUtMy4wMjkzYy4wMDMtNC4xNDQtMy4zNTYtNy41MDQ4LTcuNDk4LTcuNTA5OC0yLjQ4NCAwLTQuNS0yLjAxNi00LjUtNC41UzkuNTE2IDEuNSAxMiAxLjVzNC41IDIuMDE2IDQuNSA0LjVIMThjMC0zLjMxMi0yLjY4OC02LTYtNnptMCAzYy0xLjU5Ny4wNC0yLjg3OTkgMS4zMjU5LTIuOTIxOSAyLjkyMTlDOS4wMzUxIDcuNTc5OSAxMC4zNDMgOC45NTcgMTIgOWMxLjU5Ny0uMDQgMi44Nzk5LTEuMzI1OSAyLjkyMTktMi45MjE5QzE0Ljk2NDkgNC40MjAxIDEzLjY1NiAzLjA0MyAxMiAzem0wIDEuNWMuODI4IDAgMS41LjY3MiAxLjUgMS41di4wMDJjMCAuODI4LS42NzIgMS41LTEuNSAxLjUtLjgzIDAtMS41LS42NzMtMS41LTEuNTAyIDAtLjgzLjY3LTEuNSAxLjUtMS41ek03LjUgMTV2MS41SDl2Nkg0LjVWMjRoMTV2LTEuNUgxNXYtNmgxLjVWMTVoLTl6bTMgMS41aDN2NmgtM3YtNnptLTYgMS40NzA3VjE4YTcuNDE4IDcuNDE4IDAgMCAwIC42NDQ1IDMuMDM5aDEuNjgzNkM2LjMwMDEgMjAuMTQ3IDYgMTkuMTEgNiAxOHYtLjAyOTNINC41WiIvPjwvc3ZnPg==" />
```
```
<img src="data:image/svg+xml;base64,PHN2ZyByb2xlPSJpbWciIHZpZXdCb3g9IjAgMCAyNCAyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48dGl0bGU+U2ltcGxlIEljb25zIGljb248L3RpdGxlPjxwYXRoIGZpbGw9IiNmZmYiIGQ9Ik0xMiAwQzguNjg4IDAgNiAyLjY4OCA2IDZzMi42ODggNiA2IDZjMS4wNjYgMCAyLjExNzEuMjg2MiAzLjAzNzEuODI2MiAyLjg1OCAxLjY3OCAzLjgxNjcgNS4zNTM5IDIuMTM4NyA4LjIxMjloMS42Nzk3YTcuNDkyNSA3LjQ5MjUgMCAwIDAgLjY0MjUtMy4wMjkzYy4wMDMtNC4xNDQtMy4zNTYtNy41MDQ4LTcuNDk4LTcuNTA5OC0yLjQ4NCAwLTQuNS0yLjAxNi00LjUtNC41UzkuNTE2IDEuNSAxMiAxLjVzNC41IDIuMDE2IDQuNSA0LjVIMThjMC0zLjMxMi0yLjY4OC02LTYtNnptMCAzYy0xLjU5Ny4wNC0yLjg3OTkgMS4zMjU5LTIuOTIxOSAyLjkyMTlDOS4wMzUxIDcuNTc5OSAxMC4zNDMgOC45NTcgMTIgOWMxLjU5Ny0uMDQgMi44Nzk5LTEuMzI1OSAyLjkyMTktMi45MjE5QzE0Ljk2NDkgNC40MjAxIDEzLjY1NiAzLjA0MyAxMiAzem0wIDEuNWMuODI4IDAgMS41LjY3MiAxLjUgMS41di4wMDJjMCAuODI4LS42NzIgMS41LTEuNSAxLjUtLjgzIDAtMS41LS42NzMtMS41LTEuNTAyIDAtLjgzLjY3LTEuNSAxLjUtMS41ek03LjUgMTV2MS41SDl2Nkg0LjVWMjRoMTV2LTEuNUgxNXYtNmgxLjVWMTVoLTl6bTMgMS41aDN2NmgtM3YtNnptLTYgMS40NzA3VjE4YTcuNDE4IDcuNDE4IDAgMCAwIC42NDQ1IDMuMDM5aDEuNjgzNkM2LjMwMDEgMjAuMTQ3IDYgMTkuMTEgNiAxOHYtLjAyOTNINC41WiIvPjwvc3ZnPg==" />
```
That would give
![image](https://user-images.githubusercontent.com/12817388/112840283-24502800-906d-11eb-99ec-c2ec44ef0d20.png)

Reply:
I "solved" it, though it required CSS. My initial idea was to use an approach similar to what you can find in [this article](https://flaviocopes.com/dark-mode-change-img-source/), but that doesn't work with the _forced_ dark mode and light mode :disappointed: 

If your browser supports lazy loading and JS is enabled, this solution works great. It only loads the dark mode icons once you go into dark mode - and again the cache is used whenever possible.

The downside is that if neither of these are true, the browser will end up downloading **all** SVGs **twice** when you load the website which is really, really bad...

Reply:
> The possible alternative would be to use `img` with data url (svg encoded in base64). Something like this:
> 
> ```
> <img src="data:image/svg+xml;base64,PHN2ZyByb2xlPSJpbWciIHZpZXdCb3g9IjAgMCAyNCAyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48dGl0bGU+U2ltcGxlIEljb25zIGljb248L3RpdGxlPjxwYXRoIGQ9Ik0xMiAwQzguNjg4IDAgNiAyLjY4OCA2IDZzMi42ODggNiA2IDZjMS4wNjYgMCAyLjExNzEuMjg2MiAzLjAzNzEuODI2MiAyLjg1OCAxLjY3OCAzLjgxNjcgNS4zNTM5IDIuMTM4NyA4LjIxMjloMS42Nzk3YTcuNDkyNSA3LjQ5MjUgMCAwIDAgLjY0MjUtMy4wMjkzYy4wMDMtNC4xNDQtMy4zNTYtNy41MDQ4LTcuNDk4LTcuNTA5OC0yLjQ4NCAwLTQuNS0yLjAxNi00LjUtNC41UzkuNTE2IDEuNSAxMiAxLjVzNC41IDIuMDE2IDQuNSA0LjVIMThjMC0zLjMxMi0yLjY4OC02LTYtNnptMCAzYy0xLjU5Ny4wNC0yLjg3OTkgMS4zMjU5LTIuOTIxOSAyLjkyMTlDOS4wMzUxIDcuNTc5OSAxMC4zNDMgOC45NTcgMTIgOWMxLjU5Ny0uMDQgMi44Nzk5LTEuMzI1OSAyLjkyMTktMi45MjE5QzE0Ljk2NDkgNC40MjAxIDEzLjY1NiAzLjA0MyAxMiAzem0wIDEuNWMuODI4IDAgMS41LjY3MiAxLjUgMS41di4wMDJjMCAuODI4LS42NzIgMS41LTEuNSAxLjUtLjgzIDAtMS41LS42NzMtMS41LTEuNTAyIDAtLjgzLjY3LTEuNSAxLjUtMS41ek03LjUgMTV2MS41SDl2Nkg0LjVWMjRoMTV2LTEuNUgxNXYtNmgxLjVWMTVoLTl6bTMgMS41aDN2NmgtM3YtNnptLTYgMS40NzA3VjE4YTcuNDE4IDcuNDE4IDAgMCAwIC42NDQ1IDMuMDM5aDEuNjgzNkM2LjMwMDEgMjAuMTQ3IDYgMTkuMTEgNiAxOHYtLjAyOTNINC41WiIvPjwvc3ZnPg=="  loading="lazy" />
> ```
> 
> ```
> <img src="data:image/svg+xml;base64,PHN2ZyByb2xlPSJpbWciIHZpZXdCb3g9IjAgMCAyNCAyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48dGl0bGU+U2ltcGxlIEljb25zIGljb248L3RpdGxlPjxwYXRoIGZpbGw9IiNmZmYiIGQ9Ik0xMiAwQzguNjg4IDAgNiAyLjY4OCA2IDZzMi42ODggNiA2IDZjMS4wNjYgMCAyLjExNzEuMjg2MiAzLjAzNzEuODI2MiAyLjg1OCAxLjY3OCAzLjgxNjcgNS4zNTM5IDIuMTM4NyA4LjIxMjloMS42Nzk3YTcuNDkyNSA3LjQ5MjUgMCAwIDAgLjY0MjUtMy4wMjkzYy4wMDMtNC4xNDQtMy4zNTYtNy41MDQ4LTcuNDk4LTcuNTA5OC0yLjQ4NCAwLTQuNS0yLjAxNi00LjUtNC41UzkuNTE2IDEuNSAxMiAxLjVzNC41IDIuMDE2IDQuNSA0LjVIMThjMC0zLjMxMi0yLjY4OC02LTYtNnptMCAzYy0xLjU5Ny4wNC0yLjg3OTkgMS4zMjU5LTIuOTIxOSAyLjkyMTlDOS4wMzUxIDcuNTc5OSAxMC4zNDMgOC45NTcgMTIgOWMxLjU5Ny0uMDQgMi44Nzk5LTEuMzI1OSAyLjkyMTktMi45MjE5QzE0Ljk2NDkgNC40MjAxIDEzLjY1NiAzLjA0MyAxMiAzem0wIDEuNWMuODI4IDAgMS41LjY3MiAxLjUgMS41di4wMDJjMCAuODI4LS42NzIgMS41LTEuNSAxLjUtLjgzIDAtMS41LS42NzMtMS41LTEuNTAyIDAtLjgzLjY3LTEuNSAxLjUtMS41ek03LjUgMTV2MS41SDl2Nkg0LjVWMjRoMTV2LTEuNUgxNXYtNmgxLjVWMTVoLTl6bTMgMS41aDN2NmgtM3YtNnptLTYgMS40NzA3VjE4YTcuNDE4IDcuNDE4IDAgMCAwIC42NDQ1IDMuMDM5aDEuNjgzNkM2LjMwMDEgMjAuMTQ3IDYgMTkuMTEgNiAxOHYtLjAyOTNINC41WiIvPjwvc3ZnPg=="  loading="lazy" />
> ```
> 
> That would give
> ![image](https://user-images.githubusercontent.com/12817388/112840283-24502800-906d-11eb-99ec-c2ec44ef0d20.png)

Sorry, missed this comment. That defeats the entire goal of what we're trying to do here, which is lazy loading the SVGs. If we provide the SVG source directly in the `src` attribute of the SVG we might as well embed the SVG itself - which is what we already have on `master`.

Reply:
> The downside is that if neither of these are true, the browser will end up downloading all SVGs twice when you load the website which is really, really bad...

Yep, that is why I was trying to fin another alternative.

> That defeats the entire goal of what we're trying to do here, which is lazy loading the SVGs.

I think SVGs will still be « lazy rendered », so it might not defeats entirely the goal of this PR. But I understand that the HTML will still be large. Would surely need some test since the behavior of lazy loading inline images is not well documented.

Reply:
I might have a compromise.
If you only add the default black svg and apply the following style on dark mode, it might do it:
```css
    filter: invert(1);
```
The support is not bad (https://caniuse.com/css-filters)

Reply:
It has better support than `loading="lazy"` itself so worth a try, but indeed it's less than ideal.

Reply:
> I think SVGs will still be « lazy rendered », so it might not defeats entirely the goal of this PR. But I understand that the HTML will still be large. Would surely need some test since the behavior of lazy loading inline images is not well documented.

It does indeed seem to be the case that the icons are still "lazy rendered" and it does indeed seem to still improve performance. I implemented this approach in ba07d08dbba9aa3dc138c7681e55d09af808b003.